code/datums/emotes.dm

```diff
@@ -0,0 +1,165 @@
+#define EMOTE_VISIBLE 1
+#define EMOTE_AUDIBLE 2
+
+/datum/emote
+	var/key = "" //What calls the emote
+	var/key_third_person = "" //This will also call the emote
+	var/message = "" //Message displayed when emote is used
+	var/message_param = "" //Message to display if a param was given
+	var/emote_type = EMOTE_VISIBLE //Whether the emote is visible or audible
+	var/restraint_check = FALSE //Checks if the mob is restrained before performing the emote
+	var/muzzle_ignore = FALSE //Will only work if the emote is EMOTE_AUDIBLE
+	var/muzzled_noise = "" // For snowflaking the muzzled noise
+	var/list/mob_type_blacklist_typecache //Types that are NOT allowed to use that emote
+	var/list/mob_type_ignore_stat_typecache
+	var/punct = "."
```

Comment:
This really shouldn't be necessary.

Reply:
I honestly think it's better to include punctuation in the emote, cause it's useful for automatically generating directional emote variations. Otherwise you have to do a bit of work to identify the punctuation used, but if you're expecting punctuation at the end of emotes anyways, then it should be coded in